mrmustard/lab/states/dm.py

```diff
@@ -292,13 +292,34 @@ def expectation(self, operator: CircuitComponent, mode: str = "kron") -> Batch[S
             raise ValueError(msg)
 
         leftover_modes = self.wires.modes - operator.wires.modes
+        # custom shape handling from contract
+        # since input and output wires are contracted
+        # to do the trace out
+        if (
+            type(self.ansatz) is not type(operator.ansatz)
+            and settings.DEFAULT_REPRESENTATION == "Fock"
+        ):
+            self_shape = list(self.auto_shape())
+            other_shape = list(operator.auto_shape())
+            # want to make sure that only the operator modes use shape lookahead
+            # for efficiency
+            for m in operator.modes:
+                for idx1, idx2 in zip(self.wires[m].indices, operator.wires[m].indices):
+                    max_shape = max(self_shape[idx1], other_shape[idx2])
+                    self_shape[idx1] = max_shape
+                    other_shape[idx2] = max_shape
+            self_rep = self.to_fock(tuple(self_shape))
+            other_rep = operator.to_fock(tuple(other_shape))
+        else:
+            self_rep = self
+            other_rep = operator
```

Comment:
Nit:
```suggestion
            self_shape = list(self.auto_shape())
            op_shape = list(operator.auto_shape())
            # want to make sure that only the operator modes use shape lookahead
            # for efficiency
            for m in operator.modes:
                for idx1, idx2 in zip(self.wires[m].indices, operator.wires[m].indices):
                    max_shape = max(self_shape[idx1], op_shape[idx2])
                    self_shape[idx1] = max_shape
                    op_shape[idx2] = max_shape
            self_rep = self.to_fock(tuple(self_shape))
            op_rep = operator.to_fock(tuple(op_shape))
        else:
            self_rep = self
            op_rep = operator
```
or perhaps the more verbose "operator"? Just seems a little odd to call it "other" when it's passed as `operator`.